src/TrackPanel.cpp

```diff
@@ -524,8 +524,9 @@ void TrackPanel::MakeParentRedrawScrollbars()
 namespace {
    std::shared_ptr<Track> FindTrack(TrackPanelCell *pCell )
    {
-      if (pCell)
-         return static_cast<CommonTrackPanelCell*>( pCell )->FindTrack();
+      if (auto pCommonCell = dynamic_cast<CommonTrackPanelCell*>(pCell))
```

Comment:
Why was this change needed?  Because of EmptyPanelRect not inheritinc CommonTrackPanelCell?

Reply:
It casts to the subclass, so static_cast is not safe.

Reply:
@crsib that's not a complete answer.  static_cast can be safe in special cases where the programmer has some other way to prove that the cast is always correcct.

The static_cast was a safe thing before that was in the code for years.  I want to understand, why is it no longer so.

Reply:
If it's guranteed to be a pointer to CommonTrackPanelCell, why then we use TrackPanelCell instead?
And in this case the problem is that EmptyPanelRect class is not derived from CommonTrackPanelCell, and the signature of TrackPanelGroup::Children doesn't say it has to be

Reply:
The complete answer why the old code was not wrong to use static_cast:  you can examine the subdivision of the area in TrackPanel, and prove that the ultimate cells were always of subclasses of CommonTrackPanelCell.  That's not true in general for a CellularPanel (the base class of TrackPanel), but it is true for this subclass of it.

This is no longer true after these changes.  It might be made true again if EmptyPanelRect could also be a CommonTrackPanelCell, knowing which track it is associated with (at construction time), and overriding DoFindTrack() to return that track.

In fact it might be mildly advantageous to rewrite that class this way.  Then the static_cast can be restored.

Reply:
> If it's guranteed to be a pointer to CommonTrackPanelCell, why then we use TrackPanelCell instead?
> And in this case the problem is that EmptyPanelRect class is not derived from CommonTrackPanelCell, and the signature of TrackPanelGroup::Children doesn't say it has to be

Good points:  but TrackPanelGroup is misleadingly named for historical reasons (and my fault): CellularPanelGroup would be a better name.  It is now meant to cooperate with CellularPanel, a more general base class of TrackPanel but also of AdornedRulerPanel (the timeline above it).

"TrackPanelCell" might be better called CellularPanelCell, and "CommonTrackPanelCell" is really the common subclass of all cells, specifically of TrackPanel, so "TrackPanelCell" should be the name only for that.

The guarantee that the pointer to the visited cell is really of this subclass -- that guarantee is not given by CellularPanel, but it IS given by TrackPanel.

Did I make that clear?  I'm not sure I did, but that's my explanation.

Reply:
You are saying that there is an agreement that TrackPanel should contain only entities derived from CommonTrackPanelCell. I've made a small research and found out that these are not the only places where such assumption is used, so i believe it would be safer to do what you said. Refactoring of this, if required at all, is outside of the scope of this PR.